From ce25d0b144210083012542b5df7c7bf9dee3df5c Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 15 Jul 2015 13:32:06 +0200
Subject: [PATCH] Abort connection when SSL verification fails

Fix a security issue -- links would not verify SSL and TLS
certificates, rendering SSL support practically useless.

http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=510417
---
 connect.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/connect.c b/connect.c
index 0c1feb0..b532d58 100644
--- a/connect.c
+++ b/connect.c
@@ -679,6 +679,8 @@ static void connected(struct connection
 			}
 		}
 #endif
+		if (!getenv("UNSAFE_SSL"))
+			SSL_set_verify(c->ssl, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, NULL);
 		SSL_set_fd(c->ssl, *b->sock);
 		ssl_setup_downgrade(c);
 #ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
